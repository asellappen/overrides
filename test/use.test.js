import useOverrides from "../src/use";

import React from "react";
import renderer from "react-test-renderer";

const Bar = props => <div {...props}>Bar</div>;
const Baz = props => <div {...props}>Baz</div>;
const defaultComponents = { Bar };
const Foo = props => {
  const { Bar } = useOverrides(props, defaultComponents);
  return <Bar {...props} />;
};

describe("useOverrides", () => {
  it("should render without overrides", () => {
    const component = renderer.create(<Foo />);

    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should render with style overrides", () => {
    const component = renderer.create(
      <Foo overrides={{ Bar: { style: { color: "red" } } }} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should render with props overrides", () => {
    const component = renderer.create(
      <Foo overrides={{ Bar: { props: { className: "Bar" } } }} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should render with component overrides", () => {
    const component = renderer.create(
      <Foo overrides={{ Bar: { component: Baz } }} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
