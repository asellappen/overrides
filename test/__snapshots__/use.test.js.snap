// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useOverrides should render with component overrides 1`] = `
<div
  overrides={
    Object {
      "Bar": Object {
        "component": [Function],
      },
    }
  }
  style={Object {}}
>
  Baz
</div>
`;

exports[`useOverrides should render with props overrides 1`] = `
<div
  className="Bar"
  overrides={
    Object {
      "Bar": Object {
        "props": Object {
          "className": "Bar",
        },
      },
    }
  }
  style={Object {}}
>
  Bar
</div>
`;

exports[`useOverrides should render with style overrides 1`] = `
<div
  overrides={
    Object {
      "Bar": Object {
        "style": Object {
          "color": "red",
        },
      },
    }
  }
  style={
    Object {
      "color": "red",
    }
  }
>
  Bar
</div>
`;

exports[`useOverrides should render without overrides 1`] = `
<div
  style={Object {}}
>
  Bar
</div>
`;
